snuba/migrations/snuba_migrations/metrics/0020_polymorphic_buckets_table.py

```diff
@@ -0,0 +1,74 @@
+from typing import Sequence
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.utils.schemas import Array, Column, DateTime, Float, Nested, String, UInt
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Create a polymorphic bucket table for metrics_sets,
+    metrics_distributions, and metrics_counters
+    """
+
+    blocking = False
+    local_table_name = "metrics_buckets_consolidated"
+    dist_table_name = "metrics_buckets_consolidated_dist"
+
+    column_list: Sequence[Column[Modifiers]] = [
+        Column("org_id", UInt(64)),
+        Column("project_id", UInt(64)),
+        Column("metric_id", UInt(64)),
+        Column("timestamp", DateTime()),
+        Column("tags", Nested([Column("key", UInt(64)), Column("value", UInt(64))]),),
+        Column("metric_type", String(Modifiers(low_cardinality=True))),
+        Column("set_values", Array(UInt(64))),
+        Column("count_value", Float(64)),
+        Column("distribution_values", Array(Float(64))),
+        Column("materialization_version", UInt(8)),
+        Column("retention_days", UInt(16)),
+        Column("partition", UInt(16)),
+        Column("offset", UInt(64)),
+    ]
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.METRICS,
+                table_name=self.local_table_name,
+                columns=self.column_list,
+                engine=table_engines.MergeTree(
+                    storage_set=StorageSetKey.METRICS,
+                    order_by="(project_id, metric_id, tags.key, tags.value, timestamp)",
```

Comment:
Is removing org_id intentional? The existing metrics entity requires org_id be supplied in every query, and seems like a fairly efficient filter. I assume this is related to the performance tracing you've been doing.

Reply:
The thought process behind that is that there's no filtering that org_id does that project_id doesn't drill down better. If we want to add org_id into the ORDER BY I'm fine with it, but I think it's a less effective ordering column than project_id

Reply:
Ok a few changes. 
- `metric_type` should go first. The matview will be much more efficient as it always filter by metric type 
- remove tags.key/tags.value. This is a mergetree, not a RreplacingMergeTree or aggregatingMergeTree, so rows do not get squashed together. Adding the tags.key/tags.value only makes the index unnecessarily big. tags can be somewhere around 4 * 2 * 64 bit
- I would still keep the org_id for as long as it is in the aggregated tables. It is small.